keps/sig-windows/4802-windows-node-shutdown/README.md

```diff
@@ -0,0 +1,541 @@
+        kubeletConfig.ShutdownGracePeriod
+
+### Graduation Criteria
+
+
+#### Alpha Graduation
+
+* Implemented the feature for Windows service only
+
+* Investigate how e2e tests can be implemented (e.g. may need to create fake
+  shutdown event)
+
+#### Alpha -> Beta Graduation
+
+* Addresses feedback from alpha testers
+* Sufficient E2E and unit testing
+
+#### Beta -> GA Graduation
+
+* Addresses feedback from beta
+* Sufficient number of users using the feature
+* Confident that no further API / kubelet config configuration options changes are needed
+* Close on any remaining open issues & bugs
+
+### Upgrade / Downgrade Strategy
+
+Before GA graduation, feature gates will be used to control the enable/disable of this enhacement. During all phase, 
+in order to enable this feature, the kubelet itself will need to run as a windows service.
+
+### Version Skew Strategy
+
+n/a
+
+## Production Readiness Review Questionnaire
+
+
+### Feature Enablement and Rollback
+
+_This section must be completed when targeting alpha to a release._
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [X] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: `WindowsGracefulNodeShutdown`
+    - Components depending on the feature gate:
+      - `kubelet`
+  - [ ] Other
+    - Describe the mechanism:
+    - Will enabling / disabling the feature require downtime of the control
+      plane?
+      - no
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+      - yes (will require restart of kubelet)
+
+* **Does enabling the feature change any default behavior?**
+  Any change of default behavior may be surprising to users or break existing
+  automations, so be extremely careful here.
+
+    * The main behavior change is that during a node shutdown, pods running on
+      the node will be terminated gracefully.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Describe the consequences on existing workloads (e.g., if this is a runtime
+  feature, can it break the existing applications?).
+
+    * Yes, the feature can be disabled by either disabling the feature gate, or
+      setting `kubeletConfig.ShutdownGracePeriod` to 0 seconds.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+
+    * Kubelet will attempt to perform graceful termination of pods during a
+        node shutdown.
+
+* **Are there any tests for feature enablement/disablement?**
+  The e2e framework does not currently support enabling or disabling feature
+  gates. However, unit tests in each component dealing with managing data, created
+  with and without the feature, are necessary. At the very least, think about
+  conversion tests if API types are being modified.
+
+    *   n/a
```

Comment:
I don't see how this feature could be meaningfully tested without an e2e.

Reply:
we do have e2e test, please refer to the attached code PR link, will add more clarification for this